unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,152 @@
+import json
+from dataclasses import dataclass, field
+from datetime import timedelta
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from unstructured_ingest.enhanced_dataclass import enhanced_field
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+)
+
+if TYPE_CHECKING:
+    from couchbase.cluster import Cluster
+
+CONNECTOR_TYPE = "couchbase"
+SERVER_API_VERSION = "1"
+
+
+@dataclass
+class CouchbaseAccessConfig(AccessConfig):
+    connection_string: Optional[str] = None
+    username: Optional[str] = None
+    password: str = enhanced_field(sensitive=True, default=None)
+
+    def __post_init__(self):
+        if not self.connection_string and not (self.username and self.password):
+            raise ValueError("Either connection string or username and password must be set")
+
+
+@dataclass
+class CouchbaseConnectionConfig(ConnectionConfig):
+    access_config: CouchbaseAccessConfig = enhanced_field(default_factory=CouchbaseAccessConfig)
+    bucket: Optional[str] = None
+    scope: Optional[str] = None
+    collection: Optional[str] = None
+    batch_size: int = 50
+    connector_type: str = CONNECTOR_TYPE
```

Comment:
Also are `bucket`, `scope` or `collection` required? If so, let's not mark then as `Optional` with a default value set. 
```suggestion
@dataclass
class CouchbaseConnectionConfig(ConnectionConfig):
    connection_string: str
    username: str
    access_config: CouchbaseAccessConfig = enhanced_field(default_factory=CouchbaseAccessConfig)
    bucket: Optional[str] = None
    scope: Optional[str] = None
    collection: Optional[str] = None
    batch_size: int = 50
    connector_type: str = CONNECTOR_TYPE
```

Reply:
I have added few defaults so that it becomes a bit easier for a local couchbase cluster. However, in most cases, these values are needed to be overriden